agent/consul/leader_peering.go

```diff
@@ -101,7 +108,22 @@ func (s *Server) emitPeeringMetricsOnce(logger hclog.Logger, metricsImpl *metric
 			labels = append(labels, metrics.Label{Name: "partition", Value: part})
 		}
 
+		// exported services count metric
+		esc := status.GetExportedServicesCount()
 		metricsImpl.SetGaugeWithLabels(leaderExportedServicesCountKey, float32(esc), labels)
+
+		// peering health metric
+		if !status.Connected {
```

Comment:
I've added a `NeverConnected` prop to the `Status` of a stream 🤔 . I'm not sure if we'd want to have this property be deduced from `Connected bool` and `DisconnectTime time` but that could get a bit messy.

Reply:
Also, regarding the tri state ` { 1 , 0, -1 } `, I went with `NaN` given other health metrics (like `autopilot.is_healthy`) but I am not 100% sold on it. I don't really like `NaN`s because they can be hard to interpret.

Reply:
I'm coming here late but this PR caused some of my PR's tests to fail so I had to poke around.

I think `IsHealthy` being dependent on `Connected` (or conversely, `DisconnectTime`) should be okay because the retry loops will continuously retry peering connections and smooth over transient connection errors.

Reply:
Made https://github.com/hashicorp/consul/pull/14371 as a followup. Let me know if the changes make sense